### PR TITLE
Deprecate checkmax/max_params docs

### PR DESCRIPTION
**Pull request by jfy133**
It would be great someone can do a double check we've not left anything in.

We should also think about _when_ this PR should be merged in... is it OK to do this now or should we sync with the tools update/release?

### CODE REVIEW

File: sites/main-site/public/_redirects [hunk 1]
```diff
@@ -40,7 +40,7 @@
 /docs/usage/offline                                          https://nf-core-docs.netlify.app/docs/usage/getting_started/offline 200
 /docs/usage/troubleshooting                                  https://nf-core-docs.netlify.app/docs/usage/troubleshooting/overview 200
 /docs/usage/tutorials                                        https://nf-core-docs.netlify.app/docs/tutorials 200
-/docs/usage/tutorials/step_by_step_institutional_profile     https://nf-core-docs.netlify.app/docs/tutorials/use_nf-core_pipelines/config_institutional_profile 200
+/docs/usage/tutorials/step_by_step_institutional_profile     https://nf-core-docs.netlify.app/docs/tutorials/use_nf-core_pipelines/writing_institutional_profile 200
```
Comment (ewels) [inline sites/main-site/public/_redirects:43]: Should also add a new redirection rule as well if we are removing `/config_institutional_profile`

File: sites/docs/src/content/docs/tutorials/external_usage/nf-core_configs_outside_nf-core.md [hunk 2]
```diff
@@ -7,84 +7,126 @@ When you've been working with nf-core pipelines for a while, you often will get
...
+
+   :::
+
+   For a more sophisticated `base.config`, see the full [nf-core template](https://github.com/nf-core/tools/blob/master/nf_core/pipeline-template/conf/base.config)
+
+2. In a top level `nextflow.config`, specify two `param` to specify the URL where to look for nf-core/configs from.
+
+   ```groovy
+   custom_config_version      = 'master'
+   custom_config_base         = "https://raw.githubusercontent.com/nf-core/configs/${params.custom_config_version}"
```
Comment (mirpedrol) [inline sites/docs/src/content/docs/tutorials/external_usage/nf-core_configs_outside_nf-core.md:93]: I would write the `params` scope or write it as `params.custom_config_version` to make it clear
```suggestion
   params {
     custom_config_version      = 'master'
     custom_config_base         = "https://raw.githubusercontent.com/nf-core/configs/${params.custom_config_version}"
   }
```

File: sites/docs/src/content/docs/tutorials/external_usage/nf-core_configs_outside_nf-core.md [hunk 3]
```diff
@@ -7,84 +7,126 @@ When you've been working with nf-core pipelines for a while, you often will get
...
+         }
+     }
+   }
+   ```
+
+   :::
+
+   For a more sophisticated `base.config`, see the full [nf-core template](https://github.com/nf-core/tools/blob/master/nf_core/pipeline-template/conf/base.config)
+
+2. In a top level `nextflow.config`, specify two `param` to specify the URL where to look for nf-core/configs from.
```
Comment (mirpedrol) [inline sites/docs/src/content/docs/tutorials/external_usage/nf-core_configs_outside_nf-core.md:89]:
```suggestion
2. In a top level `nextflow.config`, specify two `params` to specify the URL where to look for nf-core/configs from.
```

File: sites/docs/src/content/docs/usage/Getting_started/configuration.md [hunk 4]
```diff
@@ -112,18 +112,42 @@ For more information on how to specify an executor, please refer to the [Nextflo
...
+process {
+  resourceLimits = [
+    cpus: 32,
+    memory: 256.GB,
+    time: 24.h
+  ]
+}
+```
+
+And supply this in your pipeline run command with `-c <custom>.conf`. Then, during a pipeline run, if for example a job exceeds the default memory request, it will be retried increasing the memory each time until either the job completes, or until it reaches a request of `256.GB`.
```
Comment (mirpedrol) [inline sites/docs/src/content/docs/usage/Getting_started/configuration.md:129]:
```suggestion
And supply this in your pipeline run command with `-c <custom>.config`. Then, during a pipeline run, if for example a job exceeds the default memory request, it will be retried increasing the memory each time until either the job completes, or until it reaches a request of `256.GB`.
```

File: sites/main-site/public/_redirects [hunk 5]
```diff
@@ -40,7 +40,8 @@
...
+/docs/usage/tutorials/config_institutional_profile           https://nf-core-docs.netlify.app/docs/tutorials/use_nf-core_pipelines/writing_institutional_profile 200
```
Comment (jfy133) [inline sites/main-site/public/_redirects:44]:
```suggestion
/docs/usage/tutorials/use_nf-core_pipelines/config_institutional_profile           https://nf-core-docs.netlify.app/docs/tutorials/use_nf-core_pipelines/writing_institutional_profile 200
```

File: sites/main-site/public/_redirects (hunk 1 above)
Comment (jfy133) [reply sites/main-site/public/_redirects:43]: Like so @ewels ? (see below)

File: sites/docs/src/content/docs/tutorials/external_usage/nf-core_configs_outside_nf-core.md [hunk 6]
```diff
@@ -7,84 +7,128 @@ When you've been working with nf-core pipelines for a while, you often will get
 
 One such thing is the centralised nf-core/configs repository of pre-configured configuration files that allow nf-core pipelines to run optimally on institutional clusters via the `-profile` parameter, e.g. `-profile uppmax`. A list of existing institutional profiles can be seen on the [nf-core website](https://nf-co.re/configs).
 
+:::tip
+If you want to write your own instituional profile, see the [guide on how to write a new institutional profile](/docs/tutorials/use_nf-core_pipelines/writing_institutional_profiles).
```
Comment (christopher-hakkaart) [inline sites/docs/src/content/docs/tutorials/external_usage/nf-core_configs_outside_nf-core.md:11]:
```suggestion
If you want to write your own institutional profile, see the [guide on how to write a new institutional profile](/docs/tutorials/use_nf-core_pipelines/writing_institutional_profiles).
```

File: sites/docs/src/content/docs/tutorials/external_usage/nf-core_configs_outside_nf-core.md [hunk 7]
```diff
@@ -7,84 +7,128 @@ When you've been working with nf-core pipelines for a while, you often will get
...
+       memory = 7.GB
+       time   = 4.h
+   }
+   ```
+
+   (or other sensible default values)
+
+   :::note{collapse title="Note on older nf-core template/Nextflow versions"}
+
+   If you wish to use pipelines generated with the pre-`v3.0.0` nf-core template, and/or when running with Nextflow versions earlier than 24.04.0 you may need to use an older syntax for setting resources. The following prevents the resources from exceeding a maximum limit.
```
Comment (christopher-hakkaart) [inline sites/docs/src/content/docs/tutorials/external_usage/nf-core_configs_outside_nf-core.md:34]:
```suggestion
   When running pipelines generated with the nf-core template pre-version v3.0.0 or with Nextflow pre-version 24.04.0 you may need to use an older syntax for setting resources. The following closures prevent resources from exceeding a maximum limit.
```

**Comment by jfy133** [comment]
Dunno what's wrong with the tests @mashehu @ewels  :(

**Comment by mashehu** [comment]
playwright test can be ignored atm

File: sites/docs/src/content/docs/tutorials/external_usage/nf-core_configs_outside_nf-core.md (hunk 7 above)
Comment (christopher-hakkaart) [reply sites/docs/src/content/docs/tutorials/external_usage/nf-core_configs_outside_nf-core.md:34]: I found this quite hard to follow, feel free to ignore if you disagree.

File: sites/docs/src/content/docs/usage/troubleshooting/retries.md [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+Some of these errors have common causes, which allows us to provide solutions to these problem.
+
+A common issue is a tool requiring more memory than is (initially) made available to it based on the default memory specifications set in the pipeline.
+Such errors (out of memory, or OOM) are often identified by exit code `104`, and those falling between `130` - `145`.
+
+Therefore all nf-core pipelines [by default will retry](https://github.com/nf-core/tools/blob/930ece572bf23b68c7a7c5259e918a878ba6499e/nf_core/pipeline-template/conf/base.config#L18) a process if it hits one of those exit codes, but requesting more resources (memory, CPUs, and time) for the re-submitted job.
+
+All other exit codes will cause the pipeline to fail immediately, and will not be retried.
+
+However some pipelines may extend this list, or provide different retry conditions, based on the behaviour of the specific tools in the used in the pipeline.
```
Comment (christopher-hakkaart) [inline sites/docs/src/content/docs/usage/troubleshooting/retries.md:15]:
```suggestion
However, some pipelines may extend this list or provide different retry conditions based on the behaviour of the specific tools used in the pipeline.
```

File: sites/docs/src/content/docs/usage/Getting_started/configuration.md [hunk 9]
```diff
@@ -182,12 +237,12 @@ process {
 }
 ```
 
-You can be more specific than this by targeting a given process name instead of it's label using `withName`. You can see the process names in your console log when the pipeline is running For example:
+You can also be more specific than this by targeting a given _process_ (job) name instead of it's label using `withName`. You can see the process names in your console log when the pipeline is running For example:
```
Comment (christopher-hakkaart) [inline sites/docs/src/content/docs/usage/Getting_started/configuration.md:240]:
```suggestion
You can also be more specific than this by targeting a given _process_ (job) name instead of its label using `withName`. You can see the process names in your console log when the pipeline is running For example:
```

File: sites/docs/src/content/docs/usage/Getting_started/configuration.md [hunk 10]
```diff
@@ -138,7 +162,38 @@ Where possible we try to get tools to make use of the resources available, for e
...
+  }
+  withLabel:process_high {
+    cpus   = { 12 * task.attempt }
+    memory = { 84.GB * task.attempt }
+    time   = { 10.h * task.attempt }
+  }
+}
+```
+
+The `resourceLimits` list sets absolute maximums any pipeline job can request (typically corresponding to the maximum available resources on your machine). The label blocks indicate the initial 'default' resources a pipeline job will request. For most nf-core pipelines, if a job runs out of memory, it will re-try the job but increasing the amount resource requested up to the `resourceLimits` maximum.
```
Comment (christopher-hakkaart) [inline sites/docs/src/content/docs/usage/Getting_started/configuration.md:192]:
```suggestion
The `resourceLimits` list sets the absolute maximums any pipeline job can request (typically corresponding to the maximum available resources on your machine). The label blocks indicate the initial 'default' resources a pipeline job will request. For most nf-core pipelines, if a job runs out of memory, it will retry the job but increase the amount of resource requested up to the `resourceLimits` maximum.
```

File: sites/docs/src/content/docs/usage/Getting_started/configuration.md [hunk 11]
```diff
@@ -112,18 +112,42 @@ For more information on how to specify an executor, please refer to the [Nextflo
...
+process {
+  resourceLimits = [
+    cpus: 32,
+    memory: 256.GB,
+    time: 24.h
+  ]
+}
+```
+
+And supply this in your pipeline run command with `-c <custom>.config`. Then, during a pipeline run, if for example a job exceeds the default memory request, it will be retried increasing the memory each time until either the job completes, or until it reaches a request of `256.GB`.
```
Comment (christopher-hakkaart) [inline sites/docs/src/content/docs/usage/Getting_started/configuration.md:129]:
```suggestion
And supply this in your pipeline run command with `-c <custom>.config`. Then, during a pipeline run, if, for example, a job exceeds the default memory request, it will be retried, increasing the memory each time until either the job completes or until it reaches a request of `256.GB`.
```

File: sites/docs/src/content/docs/usage/Getting_started/configuration.md [hunk 12]
```diff
@@ -112,18 +112,42 @@ For more information on how to specify an executor, please refer to the [Nextflo
...
+
+Therefore, these parameters only act as a _cap_, to prevent Nextflow submitting a single job requesting resources more than what is possible on your system and requests getting out of hand.
```
Comment (christopher-hakkaart) [inline sites/docs/src/content/docs/usage/Getting_started/configuration.md:131]:
```suggestion
Therefore, these parameters only act as a _cap_to prevent Nextflow from submitting a single job requesting resources more than what is possible on your system and requests getting out of hand.
```

File: sites/docs/src/content/docs/usage/Getting_started/configuration.md [hunk 13]
```diff
@@ -112,18 +112,42 @@ For more information on how to specify an executor, please refer to the [Nextflo
...
+
+Note that specifying these will not _increase_ the resources available to the pipeline tasks! See [Tuning workflow resources](#tuning-workflow-resources) for this.
```
Comment (christopher-hakkaart) [inline sites/docs/src/content/docs/usage/Getting_started/configuration.md:133]:
```suggestion
Note that specifying these will not _increase_ the resources available to the pipeline tasks! See [Tuning workflow resources](#tuning-workflow-resources) for more infomation.
```

File: sites/docs/src/content/docs/usage/troubleshooting/retries.md [hunk 14]
```diff
@@ -0,0 +1,15 @@
+## Processes are retrying
+
+### Why did processes report an error but then retry?
+
+One of the nice things about Nextflow is it offers the ability to retry processes if they encounter an error and fail with certain bash [exit status](https://en.wikipedia.org/wiki/Exit_status) or codes.
+Some of these errors have common causes, which allows us to provide solutions to these problem.
```
Comment (christopher-hakkaart) [inline sites/docs/src/content/docs/usage/troubleshooting/retries.md:6]:
```suggestion
Some of these errors have common causes, which allows us to provide solutions to these problems.
```

File: sites/docs/src/content/docs/tutorials/use_nf-core_pipelines/writing_institutional_profiles.md [hunk 15]
```diff
@@ -208,13 +211,31 @@ params {
...
-  igenomes_base = '/<path>/<to>/igenomes/'
 }
 ```
 
+:::
+
 #### process scope
 
-Next, we can use the `process` scope to define which scheduler to use and associated options. Any option specified in this scope means that all processes in a pipeline will use the settings defined here.
+Next, we can use the `process` scope to define certain resource specifications, scheduler to use, and associated options. Any option specified in this scope means that all processes in a pipeline will use the settings defined here.
```
Comment (christopher-hakkaart) [inline sites/docs/src/content/docs/tutorials/use_nf-core_pipelines/writing_institutional_profiles.md:221]:
```suggestion
Next, we can use the `process` scope to define certain resource specifications, schedulers to use, and associated options. Any option specified in this scope means that all processes in a pipeline will use the settings defined here.
```

File: sites/docs/src/content/docs/tutorials/external_usage/nf-core_configs_outside_nf-core.md (hunk 7 above)
Comment (jfy133) [reply sites/docs/src/content/docs/tutorials/external_usage/nf-core_configs_outside_nf-core.md:34]: I see what you mean but I think putting nf-core/nextflow before makes `pre-XXX` read as an actual version, rather than saying beforee that.

Maybe I just replace pre- with 'before version'...

File: sites/docs/src/content/docs/tutorials/use_nf-core_pipelines/writing_institutional_profiles.md [hunk 16]
```diff
@@ -208,13 +211,31 @@ params {
...
+
+First, you should specify a `resourceLimits` list based on resource maximums of your machine or infrastructure.
```
Comment (christopher-hakkaart) [inline sites/docs/src/content/docs/tutorials/use_nf-core_pipelines/writing_institutional_profiles.md:223]:
```suggestion
First, you should specify a `resourceLimits` list based on the maximum resources available on your machine or infrastructure.
```

File: sites/docs/src/content/docs/tutorials/use_nf-core_pipelines/writing_institutional_profiles.md [hunk 17]
```diff
@@ -208,13 +211,31 @@ params {
...
+
+These values are used by nf-core pipelines to limit automatic resubmission of resource-related failed jobs to ensure submitted retries do not exceed the maximum available on your cluster.
```
Comment (christopher-hakkaart) [inline sites/docs/src/content/docs/tutorials/use_nf-core_pipelines/writing_institutional_profiles.md:225]:
```suggestion
These values are used by nf-core pipelines to limit the automatic resubmission of resource-related failed jobs to ensure submitted retries do not exceed the maximum available on your cluster.
```

File: sites/docs/src/content/docs/tutorials/use_nf-core_pipelines/writing_institutional_profiles.md [hunk 18]
```diff
@@ -208,13 +211,31 @@ params {
...
+These values should be the ones you found for the largest node of your cluster (i.e., the largest node a user's job can be submitted to).
```
Comment (christopher-hakkaart) [inline sites/docs/src/content/docs/tutorials/use_nf-core_pipelines/writing_institutional_profiles.md:226]:
```suggestion
These values should be the ones you found for the largest node of your cluster (i.e., the largest node to which a user's job can be submitted).
```

File: sites/docs/src/content/docs/tutorials/external_usage/nf-core_configs_outside_nf-core.md [hunk 19]
```diff
@@ -7,84 +7,128 @@ When you've been working with nf-core pipelines for a while, you often will get
...
+       memory = 7.GB
+       time   = 4.h
+   }
+   ```
+
+   (or other sensible default values)
+
+   :::note{collapse title="Note on older nf-core template/Nextflow versions"}
+
+   When running pipelines generated with the nf-core template pre-version v3.0.0 or with Nextflow pre-version 24.04.0 you may need to use an older syntax for setting resources. The following closures prevent resources from exceeding a maximum limit.
```
Comment (jfy133) [inline sites/docs/src/content/docs/tutorials/external_usage/nf-core_configs_outside_nf-core.md:34]:
```suggestion
   When running pipelines generated with the nf-core template before version v3.0.0 or with Nextflow before version 24.04.0 you may need to use an older syntax for setting resources. The following closures prevent resources from exceeding a maximum limit.
```

File: sites/docs/src/content/docs/tutorials/use_nf-core_pipelines/writing_institutional_profiles.md [hunk 20]
```diff
@@ -183,22 +185,23 @@ params {
...
 }
 ```
 
-Finally, if you have a common resource directory for the AWS `iGenomes` collection of reference genomes, this can can also go in the `params` scope.
+:::note{collapse title="Note on older nf-core pipelines"}
+
+If using older pipelines, you will also need to specify the `max_*` series of params (now not used in the pipeline template).
+
+These are used by nf-core pipelines to limit automatic resubmission of resource-related failed jobs to ensure submitted retries do not exceed the maximum available on your cluster.
+These values should be the ones you found for the largest node of your cluster (i.e., the largest node a user's job can be submitted to). For example:
```
Comment (christopher-hakkaart) [inline sites/docs/src/content/docs/tutorials/use_nf-core_pipelines/writing_institutional_profiles.md:204]:
```suggestion
If using older pipelines, you will also need to specify the `max_*` series of params (now not used in the pipeline template).

These are used by nf-core pipelines to limit the automatic resubmission of resource-related failed jobs to ensure submitted retries do not exceed the maximum available on your cluster.
These values should be the ones you found for the largest node of your cluster (i.e., the largest node a user's job can be submitted to). For example:
```

File: sites/docs/src/content/docs/tutorials/external_usage/nf-core_configs_outside_nf-core.md [hunk 21]
```diff
@@ -7,84 +7,128 @@ When you've been working with nf-core pipelines for a while, you often will get
...
+   ```
+
+   Note that instead of the parameters, you can just directly specify a URL, e.g.
+
+   ```groovy
+   includeConfig !System.getenv('NXF_OFFLINE') ? 'https://raw.githubusercontent.com/nf-core/configs/master/nfcore_custom.config' : "/dev/null"
+   ```
+
+   :::note{collapse title="Note on older nf-core template/Nextflow versions"}
+   If you wish to use pipelines generated with the pre-`v3.0.0` nf-core template, and/or when running with Nextflow versions earlier than 24.04.0 you may need to use an older syntax for loading the configs:
```
Comment (jfy133) [inline sites/docs/src/content/docs/tutorials/external_usage/nf-core_configs_outside_nf-core.md:121]:
```suggestion
   If you wish to use pipelines generated with the nf-core template before `v3.0.0` , and/or when running with Nextflow versions earlier than 24.04.0 you may need to use an older syntax for loading the configs:
```

File: sites/docs/src/content/docs/usage/Getting_started/configuration.md [hunk 22]
```diff
@@ -112,18 +112,42 @@ For more information on how to specify an executor, please refer to the [Nextflo
...
+
+:::note{collapse title="Note on older nf-core pipelines"}
+
+If you wish to use pipelines generated with the pre-`v3.0.0` nf-core template, and/or when running with Nextflow versions earlier than 24.04.0 you may need use a different syntax to prevent the resources from exceeding a maximum limit.
```
Comment (jfy133) [inline sites/docs/src/content/docs/usage/Getting_started/configuration.md:137]:
```suggestion
If you wish to use pipelines generated with the nf-core template before `v3.0.0`, and/or when running with Nextflow versions earlier than 24.04.0 you may need use a different syntax to prevent the resources from exceeding a maximum limit.
```

**Review by christopher-hakkaart — COMMENTED** [review]
I added some grammar suggestions for consideration.

File: sites/docs/src/content/docs/usage/Getting_started/configuration.md [hunk 23]
```diff
@@ -138,7 +162,38 @@ Where possible we try to get tools to make use of the resources available, for e
...
+    time   = { 10.h * task.attempt }
+  }
+}
+```
+
+The `resourceLimits` list sets absolute maximums any pipeline job can request (typically corresponding to the maximum available resources on your machine). The label blocks indicate the initial 'default' resources a pipeline job will request. For most nf-core pipelines, if a job runs out of memory, it will re-try the job but increasing the amount resource requested up to the `resourceLimits` maximum.
+
+:::note{collapse title="Note on older nf-core pipelines"}
+
+If you wish to use pipelines generated with the pre-`v3.0.0` nf-core template, and/or when running with Nextflow versions earlier than 24.04.0 you may need use a different syntax to prevent the resources from exceeding a maximum limit.
```
Comment (jfy133) [inline sites/docs/src/content/docs/usage/Getting_started/configuration.md:196]:
```suggestion
If you wish to use pipelines generated with the nf-core template before `v3.0.0`, and/or when running with Nextflow versions earlier than 24.04.0 you may need use a different syntax to prevent the resources from exceeding a maximum limit.
```

File: sites/main-site/public/_redirects [hunk 24]
```diff
@@ -40,7 +40,8 @@
 /docs/usage/offline                                          https://nf-core-docs.netlify.app/docs/usage/getting_started/offline 200
 /docs/usage/troubleshooting                                  https://nf-core-docs.netlify.app/docs/usage/troubleshooting/overview 200
 /docs/usage/tutorials                                        https://nf-core-docs.netlify.app/docs/tutorials 200
-/docs/usage/tutorials/step_by_step_institutional_profile     https://nf-core-docs.netlify.app/docs/tutorials/use_nf-core_pipelines/config_institutional_profile 200
+/docs/usage/tutorials/step_by_step_institutional_profile     https://nf-core-docs.netlify.app/docs/tutorials/use_nf-core_pipelines/writing_institutional_profile 200
+/docs/usage/tutorials/use_nf-core_pipelines/config_institutional_profile           https://nf-core-docs.netlify.app/docs/tutorials/use_nf-core_pipelines/writing_institutional_profile 200
```
Comment (mirpedrol) [inline sites/main-site/public/_redirects:44]: I think it's without `usage`?
```suggestion
/docs/tutorials/use_nf-core_pipelines/config_institutional_profile           https://nf-core-docs.netlify.app/docs/tutorials/use_nf-core_pipelines/writing_institutional_profile 200
```

**Review by mirpedrol — APPROVED** [review]
LGTM :)

File: sites/main-site/public/_redirects (hunk 24 above)
Comment (jfy133) [reply sites/main-site/public/_redirects:44]: Wouldn't the old website link be in a different redirect?

Comment (mashehu) [reply sites/main-site/public/_redirects:44]: ah, sorry, got confused.

**Review by mashehu — APPROVED** [review]
Ready to be merged after the release of tools 3.0.0
